File: src/main/java/com/gftour/controller/HomeController.java
package com.gftour.controller;

import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;

@Controller
public class HomeController {

    @GetMapping("/")
    public String home() {
        return "index";
    }
    
    @GetMapping("/file-search")
    public String fileSearch() {
        return "file-search";
    }
    
    @GetMapping("/new-registration")
    public String newRegistration() {
        return "new-registration";
    }
    
    @GetMapping("/agt-management")
    public String agtManagement() {
        return "agt-management";
    }
    
    @GetMapping("/customer-list")
    public String customerList() {
        return "customer-list";
    }
    
    @GetMapping("/group-guidelines")
    public String groupGuidelines() {
        return "group-guidelines";
    }
    
    @GetMapping("/settlement-report")
    public String settlementReport() {
        return "settlement-report";
    }
}

File: src/main/kotlin/com/gftour/controller/HomeController.kt
package com.gftour.controller

import org.springframework.security.core.Authentication
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.GetMapping

@Controller
class HomeController {

    @GetMapping("/")
    fun home(authentication: Authentication?): String {
        return if (authentication?.isAuthenticated == true) {
            val authorities = authentication.authorities.map { it.authority }
            when {
                authorities.contains("ROLE_ADMIN") -> "redirect:/admin/dashboard"
                authorities.contains("ROLE_MANAGER") -> "redirect:/manager/dashboard"
                authorities.contains("ROLE_AGENT") -> "redirect:/agent/dashboard"
                else -> "redirect:/viewer/dashboard"
            }
        } else {
            "redirect:/login"
        }
    }

    @GetMapping("/login")
    fun login(model: Model): String {
        return "auth/login"
    }
}